cmd/federation.go

```diff
@@ -11,17 +11,24 @@ import (
 )
 
 func main() {
+	// bootstrap logger
+	logger := log.NewLogger("main")
+
 	// Load connection string from .env file
 	err := godotenv.Load()
 	if err != nil {
-		log.Warn(fmt.Sprintf("failed to load env, %v", err))
+		logger.Warn(fmt.Sprintf("failed to load env, %v", err))
 	}
+
 	conn, err := db.Connect()
 	if err != nil {
-		log.Fatal("failed connecting to db", err)
+		logger.Fatal("failed connecting to db", err)
 	}
-	defer conn.Close()
 	db.RunMigrations(conn)
-	http.RunServer()
+	defer conn.Close()
```

Comment:
This is still missed. The defer call should happen directly after the if block that checks for `err != null`. This ensures that if a fatal error occurs in the migrations the connection gets closed properly.

Reply:
modified :)